### ISSUE TITLE
ma57 `get_factors.patch` build error

### ISSUE THREAD

**Issue by jacob-roth** [JuliaSmoothOptimizers/HSL.jl#15]
As a follow up to #12, I've been trying to build HSL's `ma57` branch, but I've been having trouble.

My steps were:

1. Download Homebrew.jl (after `rm -rf` removing it) with`Pkg.add("Homebrew")`
2. Move `hsl_ma57-5.2.0.tar.gz` into `downloads` 
3. Update the `build.jl` file with my SHA and comment out the ma97 stuff (for now)
4. `brew tap dpo/openblas` & `brew tap-pin dpo/openblas`
5. build HSL from Julia REPL with `Pkg.build("HSL")`

Step 5 above led to the following error:
```
failed process: Process(`patch -p1 -i /Users/jakeroth/.julia/v0.6/HSL/deps/downloads/get_factors.patch`, ProcessExited(2)) [2]
```
with similar output to @lrsantos11:

```
julia> Pkg.build("HSL")
INFO: Building Homebrew
INFO: Building HSL
INFO: looking for /Users/jakeroth/.julia/v0.6/HSL/deps/downloads/hsl_ma57-5.2.0.tar.gz
INFO: hsl_ma57 found
INFO: Attempting to create directory /Users/jakeroth/.julia/v0.6/HSL/deps/src
INFO: Directory /Users/jakeroth/.julia/v0.6/HSL/deps/src already exists
INFO: Attempting to create directory /Users/jakeroth/.julia/v0.6/HSL/deps
INFO: Directory /Users/jakeroth/.julia/v0.6/HSL/deps already exists
INFO: Path /Users/jakeroth/.julia/v0.6/HSL/deps/src/hsl_ma57-5.2.0 already exists
INFO: Changing directory to /Users/jakeroth/.julia/v0.6/HSL/deps/src/hsl_ma57-5.2.0
patch: **** Can't open patch file /Users/jakeroth/.julia/v0.6/HSL/deps/downloads/get_factors.patch : No such file or directory
===================================[ ERROR: HSL ]====================================

LoadError: failed process: Process(`patch -p1 -i /Users/jakeroth/.julia/v0.6/HSL/deps/downloads/get_factors.patch`, ProcessExited(2)) [2]
while loading /Users/jakeroth/.julia/v0.6/HSL/deps/build.jl, in expression starting on line 48

=====================================================================================

==================================[ BUILD ERRORS ]===================================

WARNING: HSL had build errors.

 - packages with build errors remain installed in /Users/jakeroth/.julia/v0.6
 - build the package(s) and all dependencies with `Pkg.build("HSL")`
 - build a single package by running its `deps/build.jl` script

=====================================================================================
```

What is the `get_factors.patch`? Any thoughts would be appreciated! Additionally, if I'm able to get it working, would it be useful for me to add a bit of detail to the documentation?

**Comment by lrsantos11** [comment]
@dpo made a patch on ma57 to make it work, but he cannot public. (see https://github.com/JuliaSmoothOptimizers/HSL.jl/issues/12#issuecomment-395817718).
He'll help, for sure.

**Comment by dpo** [comment]
I'm currently working with the HSL folks to remove the need for a patch... More news soon, I hope!

**Comment by jacob-roth** [comment]
Thanks! I'll stay in the loop!

**Comment by abelsiqueira** [comment]
Partially fixed by #34. The build will work without `get_factors.patch`, but some functions are not included.
